src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
```

Comment:
The "add", "set", and "remove" methods have similar boundary checks for the index parameter, and a separate method could be created to handle these checks and throw an appropriate exception.